CefSharp.Core/Internals/CefCertificateCallbackWrapper.h

```diff
@@ -0,0 +1,65 @@
+// Copyright � 2010-2016 The CefSharp Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+#pragma once
+
+#include "Stdafx.h"
+
+#include "include\cef_request_handler.h"
+#include "CefWrapper.h"
+
+namespace CefSharp
+{
+	namespace Internals
+	{
+		private ref class CefCertificateCallbackWrapper : public ISelectClientCertificateCallback, public CefWrapper
+		{
+		private:
+			MCefRefPtr<CefSelectClientCertificateCallback> _callback;
+			const CefRequestHandler::X509CertificateList& _certificateList;
+
+		public:
+			CefCertificateCallbackWrapper(CefRefPtr<CefSelectClientCertificateCallback>& callback, const CefRequestHandler::X509CertificateList& certificates)
+				: _callback(callback), _certificateList(certificates)
+			{
+
+			}
+
+			!CefCertificateCallbackWrapper()
+			{
+				_callback = NULL;
+			}
+
+			~CefCertificateCallbackWrapper()
+			{
+				this->!CefCertificateCallbackWrapper();
+
+				_disposed = true;
+			}
+
+			virtual void Select(System::Security::Cryptography::X509Certificates::X509Certificate2^ cert)
+			{
+				ThrowIfDisposed();
+				if (!cert){
+					_callback->Select(NULL);
+				}
+
+				//Need cert_name from cert
+				auto cert_name = cert->SubjectName->Name;
```

Comment:
Ok, so I tried to read the binary data and use `Encodeing` class like this,
```
auto serialNum((*it)->GetSerialNumber());

auto byteSize = serialNum->GetSize();

auto bufferByte = gcnew cli::array<Byte>(byteSize);
pin_ptr<Byte> src = &bufferByte[0]; // pin pointer to first element in arr

serialNum->GetData(static_cast<void*>(src), byteSize, 0);

UTF8Encoding^ encoding = gcnew UTF8Encoding;
auto serialStr(encoding->GetString(bufferByte));
```
but `serialStr` comes out as a messed up string (more like symbols that strings). I tried it with all the `Encoding` types [msdn link](https://msdn.microsoft.com/en-us/library/system.text.encoding(v=vs.110).aspx?cs-save-lang=1&cs-lang=cpp#code-snippet-1).

I ultimately, recreated the certificate from the certificateList in the iteration for each CefCertificate and then compared their serial number and that worked. I guess this is not a workaround.